tailwind.config.ts

```diff
@@ -1,20 +1,210 @@
-import type { Config } from 'tailwindcss'
+import type { Config } from "tailwindcss";
 
-const config: Config = {
+interface RecursiveKeyValuePair<K extends keyof any = string, V = string> {
+  [key: string]: V | RecursiveKeyValuePair<K, V>;
+}
+
+const config: Config & {
+  theme: { colors: RecursiveKeyValuePair };
+} = {
+  darkMode: ["class", '[data-theme="dark"]'],
   content: [
-    './pages/**/*.{js,ts,jsx,tsx,mdx}',
-    './components/**/*.{js,ts,jsx,tsx,mdx}',
-    './app/**/*.{js,ts,jsx,tsx,mdx}',
+    "./pages/**/*.{js,ts,jsx,tsx,mdx}",
+    "./components/**/*.{js,ts,jsx,tsx,mdx}",
+    "./app/**/*.{js,ts,jsx,tsx,mdx}",
+  ],
+  safelist: [
+    {
+      pattern: /bg-[a-z0-9-]*/,
+    },
```

Comment:
why do we need to safeList this pattern?

I'm aware it should only be used when is REALLY needed. Otherwise it will bump our performance. 

Just want to make sure we need this.

Reply:
This is here so theme colors can appear on UI page dynamically. Otherwise we have to declare every class by hand.